lib/info.js

```diff
@@ -84,7 +84,21 @@ module.exports = function getInfo(link, options, callback) {
       iurlsd     : THUMBNAIL_URL + id + '/sddefault.jpg',
       iurlmq     : THUMBNAIL_URL + id + '/mqdefault.jpg',
       iurlhq     : THUMBNAIL_URL + id + '/hqdefault.jpg',
-      iurlmaxres : THUMBNAIL_URL + id + '/maxresdefault.jpg',
+      iurlmaxres : function getiurlmaxres(iurlcallback) {
+        var iurlmaxres = THUMBNAIL_URL + id + '/maxresdefault.jpg';
+        request(iurlmaxres, function(iurlerr, iurlres, iurlbody) {
+          if (iurlerr && iurlerr.message.includes("Status code: 404")) {
+            if (!iurlcallback) {
+              return new Promise(function(r, j) { return r(null)});
```

Comment:
This is not how async works.
U wanna return a Promise wrapping the function before u do an async task like the request()